src/main/java/io/github/aquerr/koth/command/CreateArenaClassCommand.java

```diff
@@ -1,94 +1,52 @@
-////			final Optional<ItemStack> itemStack = slot.poll();
-////			if(itemStack.isPresent())
-////			{
-////				items.add(ItemStack.builder().from(itemStack.get()).build());
-////			}
-////		}
-//
-//		for(final Inventory slotInventory : inventory.getGrid().slots())
-//		{
-//			if(!(slotInventory instanceof Slot))
-//				continue;
-//
-//			final Slot slot = (Slot)slotInventory;
-//			final Optional<ItemStack> itemStack = slot.peek();
-//			if(itemStack.isPresent())
-//			{
-//				items.add(ItemStack.builder().fromContainer(itemStack.get().toContainer()).build());
-//			}
-//		}
-//
-//		for(final Inventory slotInventory : inventory.getHotbar())
-//		{
-//			if(!(slotInventory instanceof Slot))
-//				continue;
-//
-//			final Slot slot = (Slot)slotInventory;
-//			final Optional<ItemStack> itemStack = slot.peek();
-//			if(itemStack.isPresent())
-//			{
-//				items.add(ItemStack.builder().fromContainer(itemStack.get().toContainer()).build());
-//			}
-//		}
-//		return items;
-//	}
-//}
+package io.github.aquerr.koth.command;
+
+import io.github.aquerr.koth.manager.ArenaClassManager;
+import io.github.aquerr.koth.model.ArenaClass;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.TextComponent;
+import net.kyori.adventure.text.format.NamedTextColor;
+import org.spongepowered.api.command.CommandResult;
+import org.spongepowered.api.command.exception.CommandException;
+import org.spongepowered.api.command.parameter.CommandContext;
+import org.spongepowered.api.command.parameter.Parameter;
+import org.spongepowered.api.entity.living.player.server.ServerPlayer;
+import org.spongepowered.api.item.inventory.ItemStack;
+import org.spongepowered.api.item.inventory.Slot;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CreateArenaClassCommand extends AbstractCommand
+{
+    private final ArenaClassManager arenaClassManager;
+
+    public CreateArenaClassCommand(final ArenaClassManager arenaClassManager)
+    {
+        this.arenaClassManager = arenaClassManager;
+    }
+
+    @Override
+    public CommandResult execute(CommandContext context) throws CommandException {
+        final String className = context.requireOne(Parameter.string().key("name").build());
+        ServerPlayer serverPlayer = requireServerPlayer(context);
+
+        final List<ItemStack> items = getItemFromPlayer(serverPlayer);
+        final boolean didSucceed = this.arenaClassManager.addArenaClass(new ArenaClass(className, items));
+        if(!didSucceed)
+            throw new CommandException(Component.text("Something went wrong while creating the arena class..."));
+        serverPlayer.sendMessage(TextComponent.ofChildren(Component.text("Successfully created the arena class!", NamedTextColor.GOLD)));
+        return CommandResult.success();
+    }
+    private List<ItemStack> getItemFromPlayer(final ServerPlayer serverPlayer)
+    {
+        final List<ItemStack> items = new ArrayList<>();
```

Comment:
@PKNCK Tutaj można byłoby użyć LinkedList zamiast ArrayList. To w sumie taka mała  "mikro" optymalizacja więc może zostać jak jest :)